ibis/backends/pyarrow/datatypes.py

```diff
@@ -129,3 +131,14 @@ def infer_pyarrow_schema(schema: pa.Schema) -> sch.Schema:
     return sch.schema(
         [(f.name, dt.dtype(f.type, nullable=f.nullable)) for f in schema]
     )
+
```

Comment:
I think we should keep the pyarrow dependent logic under the pyarrow subpackage.

For example we register the `DataType.to_pandas` and `Schema.to_pandas` methods from the pandas backend, see https://github.com/ibis-project/ibis/blob/master/ibis/backends/pandas/client.py#L300-L301

Reply:
Yeah, this looks nicer, doing this now